### ISSUE TITLE
Local Lambda Authorizer always fails with InvalidLambdaAuthorizerResponse

### ISSUE THREAD

**Issue by zerubeus** [aws/aws-sam-cli#5072]
### Description:
When I use sam local start-api, my Lambda authorizer always fails with InvalidLambdaAuthorizerResponse even with a correct response.

### Steps to reproduce:
- Serverless Rest API resource
- Lambda authorizer
- Other Lambdas for the API
- API definition in a separate file, "openapi.yaml" imported in the template with DefinitionBody

The response from my Lambda authorizer:

```
{'principalId': 'user', 'policyDocument': {'Version': '2012-10-17', 'Statement': [{'Action': 'execute-api:Invoke', 'Effect': 'Allow', 'Resource': 'arn:aws:execute-api:us-east-1:123456789012:1234567890/dev/*/*'}]}}
```

Using the last version of SAM

**Comment by lucashuy** [comment]
Thanks for reporting this. I can replicate this behaviour with your sample response. Looks like we expect the `Resource` to be a list of strings, however a single string is completely valid too.

While we work to fix this, can the response be amended so that the `Resource` is a list containing that single ARN as a possible workaround to unblock your workflow for now?

**Comment by zerubeus** [comment]
@lucashuy you accept pull requests?

**Comment by lucashuy** [comment]
Yup, we welcome community contributions! If you wanted to give it a shot, feel free to give [our contributing guide](https://github.com/aws/aws-sam-cli/blob/develop/CONTRIBUTING.md) a read through to setup your environment.

As for the particular location where this validation is located, its primarily all in [a static method](https://github.com/aws/aws-sam-cli/blob/bab4da414b3b2414a6f7f34eb355834c73a9e93a/samcli/local/apigw/authorizers/lambda_authorizer.py#L500-L536) that validates the `Statement` part of the IAM policy. Feel free to give this a shot and we can answer any questions you might have.

**Comment by rhynix** [comment]
This was fixed in https://github.com/aws/aws-sam-cli/pull/5092 and is part of release v1.83.0. I believe this issue can be closed.

**Comment by lucashuy** [comment]
Thanks for following up on this issue, the fix for this has indeed been released in 1.83.0!
